platform_api/handlers/jobs_handler.py

```diff
@@ -572,15 +582,31 @@ async def _create_job_request_validator(
             allowed_gpu_models=gpu_models,
             allowed_tpu_resources=cluster_config.orchestrator.tpu_resources,
             cluster_name=cluster_config.name,
+            org_name=org_name,
             storage_scheme=STORAGE_URI_SCHEME,
         )
 
     def _get_cluster_config(
-        self, user_cluster_configs: Sequence[UserClusterConfig], cluster_name: str
+        self,
+        user_cluster_configs: Sequence[UserClusterConfig],
+        cluster_name: str,
+        org_name: Optional[str],
     ) -> ClusterConfig:
         for user_cluster_config in user_cluster_configs:
             if user_cluster_config.config.name == cluster_name:
-                return user_cluster_config.config
+                if org_name in user_cluster_config.orgs:
```

Comment:
may `orgs` have `None` in it?

Reply:
Yes, but it is OK. If `org_name` arg is `None`, then we are looking for direct access to this cluster. 
I'll improve the error message below.